api/v1alpha1/virtualmachine_conversion.go

```diff
@@ -207,19 +207,27 @@ func convert_v1alpha1_VmMetadata_To_v1alpha3_BootstrapSpec(
 }
 
 func convert_v1alpha3_BootstrapSpec_To_v1alpha1_VmMetadata(
-	in *vmopv1.VirtualMachineBootstrapSpec) *VirtualMachineMetadata {
+	btSpec *vmopv1.VirtualMachineBootstrapSpec, annotations map[string]string) *VirtualMachineMetadata {
 
-	if in == nil || apiequality.Semantic.DeepEqual(*in, vmopv1.VirtualMachineBootstrapSpec{}) {
+	if btSpec == nil || apiequality.Semantic.DeepEqual(*btSpec, vmopv1.VirtualMachineBootstrapSpec{}) {
 		return nil
 	}
 
-	// TODO: nextver only has a Secret bootstrap field so that's what we set in v1a1. If this was created
-	// as v1a1, we need to store the serialized object to know to set either the ConfigMap or Secret field.
+	// If this was created as v1a1, we can use the V1alpha1ConfigMapTransportAnnotation to know to set the ConfigMap field.
+	// nextVersions only support Secrets for bootstrap.
+	isV1a1MDConfigMapSet := false
+	if v, ok := annotations[vmopv1.V1alpha1ConfigMapTransportAnnotation]; ok && v == "true" {
```

Comment:
Maybe https://pkg.go.dev/strconv#ParseBool here instead (and just ignore the error). Or just have the presence of the key there be enough?

Thoughts are reducing the `isV1a1MDConfigMapSet` checks with like:

```
var secretName string
if cloudInit := in.CloudInit; cloudInit != nil {
  secretName =  cloudInit.RawCloudConfig.Name
...
}

if secretName != nil {
  // comment comment comment
  if _, ok := annotations[vmopv1.V1alpha1ConfigMapTransportAnnotation]; !ok {
     out.SecretName = secretName
  } else {
    out.ConfigMapName = secretName
  }
}

return out
```

Reply:
nice idea!